test/e2e/taints_test.go

```diff
@@ -0,0 +1,91 @@
+// +build e2e
+
+package e2e
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/test/framework"
+)
+
+func runTaintsUpgradeFlow(test *framework.ClusterE2ETest, updateVersion v1alpha1.KubernetesVersion, preUpgradeNodeTaints, postUpgradeNodeTaints map[corev1.Taint]int, clusterOpts ...framework.ClusterE2ETestOpt) {
```

Comment:
I feel to some point, we should not create new flows if unnecessary. Taints upgrade should be part of the upgrade workflow. The only difference is this adds two `test.ValidateNodeTaints` steps. To me it should be part of the `ValidateCluster`. And refactor the `ValidateCluster` method to take arbitrary validation checks (in this case, check node taints). So that we dont keep adding new upgrade flow when we simply want to validate different aspects of the cluster.

Reply:
Are you suggesting there should be a single workflow with assertions placed throughout and that these could just be made a part of that? Is that how e2e workflows are currently modeled?

Reply:
yeah I don't disagree....my motivation at this moment is to clearly validate the taints in isolation with as few additional factors as possible so I can make sure the feature works as expected

Reply:
added in https://github.com/aws/eks-anywhere/issues/1233

Reply:
E2E tests tend to be bigger with steps and intermediate assertions but I'm wondering if following that pattern would result in obnoxiously large e2e test workflows? Have you got a link to existing workflows?

Reply:
https://github.com/aws/eks-anywhere/tree/main/test/e2e

you can find the various flows related to various functionalities spread out through the tests in this dir